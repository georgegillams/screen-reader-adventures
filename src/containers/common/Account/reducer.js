import produce from 'immer';

import { logout, requestVerificationEmail } from './actions';

export const initialState = {
  loggingOut: false,
  logOutError: null,
  logOutResult: null,
  requestingVerificationEmail: false,
  requestVerificationEmailError: null,
  requestVerificationEmailResult: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case logout.REQUEST:
        draft.loggingOut = true;
        draft.logOutError = null;
        break;

      case logout.SUCCESS:
        draft.loggingOut = false;
        draft.logOutResult = payload;
        break;

      case logout.FAILURE:
        draft.loggingOut = false;
        draft.logOutError = payload;
        break;

      case requestVerificationEmail.REQUEST:
        draft.requestingVerificationEmail = true;
        draft.requestVerificationEmailError = null;
        break;

      case requestVerificationEmail.SUCCESS:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailResult = payload;
        break;

      case requestVerificationEmail.FAILURE:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailError = payload;
        break;
    }
  });

export default reducer;
