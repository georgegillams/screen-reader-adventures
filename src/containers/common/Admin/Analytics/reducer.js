import produce from 'immer';

import { load } from './actions';

export const initialState = {
  analytics: null,
  loading: false,
  loadError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.analytics = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;
    }
  });

export default reducer;
