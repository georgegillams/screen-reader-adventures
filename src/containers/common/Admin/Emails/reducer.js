import produce from 'immer';

import { load, resend } from './actions';

export const initialState = {
  emails: null,
  loading: false,
  loadError: null,

  emailToResend: null,
  resending: false,
  resendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.emails = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case resend.TRIGGER:
        draft.emailToResend = payload.emailToResend;
        break;

      case resend.REQUEST:
        draft.resending = true;
        draft.resendError = null;
        break;

      case resend.SUCCESS:
        draft.resending = false;
        break;

      case resend.FAILURE:
        draft.resending = false;
        draft.resendError = payload;
        break;
    }
  });

export default reducer;
