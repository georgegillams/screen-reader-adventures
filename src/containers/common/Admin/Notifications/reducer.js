import produce from 'immer';

import { load, create, remove } from './actions';

export const initialState = {
  notifications: null,
  loading: false,
  loadError: null,

  notificationToCreate: null,
  creating: false,
  createError: null,

  notificationToRemove: null,
  removing: false,
  removeError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.notifications = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case create.TRIGGER:
        draft.notificationToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case remove.TRIGGER:
        draft.notificationToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;
    }
  });

export default reducer;
