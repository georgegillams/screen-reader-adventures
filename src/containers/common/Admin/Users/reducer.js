import produce from 'immer';

import { load, remove, requestMagicLink, create, update } from './actions';

export const initialState = {
  users: null,
  loading: false,
  loadError: null,

  userToRemove: null,
  removing: false,
  removeError: null,

  userToLogIn: null,
  requesting: false,
  requestError: null,

  userToCreate: null,
  creating: false,
  createError: null,

  userToUpdate: null,
  updating: false,
  updateError: null,
  onUpdateSuccessCb: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.users = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case remove.TRIGGER:
        draft.userToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;

      case requestMagicLink.TRIGGER:
        draft.userToLogIn = payload;
        break;

      case requestMagicLink.REQUEST:
        draft.requesting = true;
        draft.requestError = null;
        break;

      case requestMagicLink.SUCCESS:
        draft.requesting = false;
        break;

      case requestMagicLink.FAILURE:
        draft.requesting = false;
        draft.requestError = payload;
        break;

      case create.TRIGGER:
        draft.userToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case update.TRIGGER:
        draft.userToUpdate = payload.userToUpdate;
        draft.onUpdateSuccessCb = payload.onUpdateSuccessCb;
        break;

      case update.REQUEST:
        draft.updating = true;
        draft.updateError = null;
        break;

      case update.SUCCESS:
        draft.updating = false;
        break;

      case update.FAILURE:
        draft.updating = false;
        draft.updateError = payload;
        break;
    }
  });

export default reducer;
