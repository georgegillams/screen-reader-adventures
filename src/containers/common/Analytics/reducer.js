import produce from 'immer';

import { sendAnalytic } from './actions';

export const initialState = {
  sending: false,
  sendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case sendAnalytic.TRIGGER:
        draft.analytic = payload;
        break;

      case sendAnalytic.REQUEST:
        draft.sending = true;
        draft.sendError = null;
        break;

      case sendAnalytic.SUCCESS:
        draft.sending = false;
        break;

      case sendAnalytic.FAILURE:
        draft.sending = false;
        draft.sendError = payload;
        break;
    }
  });

export default reducer;
