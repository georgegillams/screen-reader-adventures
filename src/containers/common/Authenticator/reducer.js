import produce from 'immer';
import cookie from 'react-cookies';

import { SESSION_COOKIE_KEY, COOKIE_EXPIRY_TIME } from 'helpers/storageConstants';
import { useLocalhost } from 'helpers/appConfig';

import { setUser, loadAuth } from './actions';

export const initialState = {
  loadingAuth: false,
  loadAuthError: null,
  user: undefined,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case setUser.TRIGGER:
        draft.user = payload;
        if (payload && payload.session) {
          cookie.save(SESSION_COOKIE_KEY, payload.session, {
            path: '/',
            secure: !useLocalhost,
            expires: new Date(Date.now() + COOKIE_EXPIRY_TIME),
          });
        }
        break;

      case loadAuth.REQUEST:
        draft.loadingAuth = true;
        draft.loadAuthError = null;
        break;

      case loadAuth.SUCCESS:
        draft.loadingAuth = false;
        draft.user = payload.user;
        break;

      case loadAuth.FAILURE:
        draft.loadingAuth = false;
        draft.loadAuthError = payload;
        break;
    }
  });

export default reducer;
