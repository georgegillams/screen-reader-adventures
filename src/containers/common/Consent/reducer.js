import produce from 'immer';

import { consent, deferConsent, setConsentReason, resetConsent } from './actions';
import { CONSENT_STATE_UNSET, CONSENT_STATE_ALLOWED, CONSENT_STATE_DEFERRED } from './constants';

export const initialState = {
  cookieConsent: null,
  cookieConsentReason: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case resetConsent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_UNSET;
        break;

      case consent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_ALLOWED;
        break;

      case deferConsent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_DEFERRED;
        break;

      case setConsentReason.TRIGGER:
        draft.cookieConsentReason = payload;
        break;
    }
  });

export default reducer;
