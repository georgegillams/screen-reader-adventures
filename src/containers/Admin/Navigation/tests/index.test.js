import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initialState as initialAuthenticatorState } from 'containers/common/Authenticator/reducer';

import configureStore from 'client-utils/common/redux/configure-store';

import AdminNavigationIndex from '../index';
import AdminNavigation from '../Container';

describe('<AdminNavigation />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNavigationIndex />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with initial state', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNavigation
          authenticatorState={{
            ...initialAuthenticatorState,
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNavigation
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'userName' },
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with admin user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNavigation
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'userName', admin: true },
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with auth error', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNavigation
          authenticatorState={{
            ...initialAuthenticatorState,
            loadAuthError: { error: 'error_type', errorMessage: 'Some error' },
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
