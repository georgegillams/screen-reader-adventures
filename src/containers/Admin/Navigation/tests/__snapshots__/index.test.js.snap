// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminNavigation /> should render correctly - index 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="admin-navigation__card-container"
    >
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminNavigation /> should render correctly with admin user 1`] = `
<div>
  <div
    class=""
  >
    <h1
      class="page-title__heading"
    >
      Admin
    </h1>
    <div
      class="admin-navigation__card-container"
    >
      <a
        aria-label="Debug settings"
        class="card feature-card feature-card--narrow-compact admin-navigation__card card--atomic card--not-disabled"
        href="/debug"
      >
        <div
          class="card__outer-container"
        >
          <div
            class="card__background"
          />
          <div
            class="card__content-container card__content-container--padded"
          >
            <div
              aria-hidden="true"
              class="feature-card__outer-container feature-card__outer-container--narrow-compact"
            >
              <div
                class="feature-card__date feature-card__date--narrow-compact"
              />
              <div
                class="feature-card__center-container feature-card__center-container--narrow-compact"
              >
                <div
                  class="section__outer section__outer--inherit-color feature-card__title"
                >
                  <h2
                    class="section__heading section__heading--no-padding"
                    id="debug-settings"
                  >
                    Debug settings
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        aria-label="Analytics"
        class="card feature-card feature-card--narrow-compact admin-navigation__card card--atomic card--not-disabled"
        href="/admin/analytics"
      >
        <div
          class="card__outer-container"
        >
          <div
            class="card__background"
          />
          <div
            class="card__content-container card__content-container--padded"
          >
            <div
              aria-hidden="true"
              class="feature-card__outer-container feature-card__outer-container--narrow-compact"
            >
              <div
                class="feature-card__date feature-card__date--narrow-compact"
              />
              <div
                class="feature-card__center-container feature-card__center-container--narrow-compact"
              >
                <div
                  class="section__outer section__outer--inherit-color feature-card__title"
                >
                  <h2
                    class="section__heading section__heading--no-padding"
                    id="analytics"
                  >
                    Analytics
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        aria-label="Emails"
        class="card feature-card feature-card--narrow-compact admin-navigation__card card--atomic card--not-disabled"
        href="/admin/emails"
      >
        <div
          class="card__outer-container"
        >
          <div
            class="card__background"
          />
          <div
            class="card__content-container card__content-container--padded"
          >
            <div
              aria-hidden="true"
              class="feature-card__outer-container feature-card__outer-container--narrow-compact"
            >
              <div
                class="feature-card__date feature-card__date--narrow-compact"
              />
              <div
                class="feature-card__center-container feature-card__center-container--narrow-compact"
              >
                <div
                  class="section__outer section__outer--inherit-color feature-card__title"
                >
                  <h2
                    class="section__heading section__heading--no-padding"
                    id="emails"
                  >
                    Emails
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        aria-label="Notifications"
        class="card feature-card feature-card--narrow-compact admin-navigation__card card--atomic card--not-disabled"
        href="/admin/notifications"
      >
        <div
          class="card__outer-container"
        >
          <div
            class="card__background"
          />
          <div
            class="card__content-container card__content-container--padded"
          >
            <div
              aria-hidden="true"
              class="feature-card__outer-container feature-card__outer-container--narrow-compact"
            >
              <div
                class="feature-card__date feature-card__date--narrow-compact"
              />
              <div
                class="feature-card__center-container feature-card__center-container--narrow-compact"
              >
                <div
                  class="section__outer section__outer--inherit-color feature-card__title"
                >
                  <h2
                    class="section__heading section__heading--no-padding"
                    id="notifications"
                  >
                    Notifications
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        aria-label="Users"
        class="card feature-card feature-card--narrow-compact admin-navigation__card card--atomic card--not-disabled"
        href="/admin/users"
      >
        <div
          class="card__outer-container"
        >
          <div
            class="card__background"
          />
          <div
            class="card__content-container card__content-container--padded"
          >
            <div
              aria-hidden="true"
              class="feature-card__outer-container feature-card__outer-container--narrow-compact"
            >
              <div
                class="feature-card__date feature-card__date--narrow-compact"
              />
              <div
                class="feature-card__center-container feature-card__center-container--narrow-compact"
              >
                <div
                  class="section__outer section__outer--inherit-color feature-card__title"
                >
                  <h2
                    class="section__heading section__heading--no-padding"
                    id="users"
                  >
                    Users
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<AdminNavigation /> should render correctly with auth error 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="admin-navigation__card-container"
    >
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
    </div>
  </div>
  <div
    class="subsection__outer loading-cover__overlay--content"
  >
    <h3
      class="subsection__heading subsection__heading--no-padding"
      id="this-is-taking-a-while.-maybe-something-isn't-quite-right..."
    >
      This is taking a while. Maybe something isn't quite right...
    </h3>
  </div>
</div>
`;

exports[`<AdminNavigation /> should render correctly with initial state 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="admin-navigation__card-container"
    >
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
      <div
        class="skeleton__outer skeleton__card--compact"
      >
        <div
          class="skeleton__shimmer"
          style="margin-left: -0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminNavigation /> should render correctly with user 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="section__outer"
    >
      <h2
        class="section__heading"
        id="admin-only"
      >
        Admin only
      </h2>
      <span
        class="paragraph"
      >
        You need to be logged in with an admin account to view this content.
        <br />
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/login"
        >
          Got a different admin account? Log in here.
        </a>
      </span>
    </div>
  </div>
</div>
`;
